staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -416,3 +418,19 @@ func parseTimeout(str string) time.Duration {
 func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
+
+type grpcError interface {
+	Code() grpccodes.Code
+	Error() string
+}
+
+func isTooLargeError(err error) bool {
+	if err != nil {
+		if grpcErr, ok := err.(grpcError); ok {
+			if grpcErr.Code() == grpccodes.InvalidArgument && grpcErr.Error() == "etcdserver: request is too large" {
```

Comment:
we have two message strings we check for in integration tests for content exceeding various sizes:

1. bigger than etcd storage limit: `etcdserver: request is too large`
2. bigger than grpc transport limit?: `rpc error: code = ResourceExhausted desc = trying to send message larger than max`

are we sure the error we would get back from etcd would always be 1 and not 2?

cc @jpbetz - can we add a test to etcd that marks this specific text as being depended on as an API so it doesn't get changed carelessly?

Reply:
In my testing only the first case came up, but if I artificially increase the size of the object to be much larger, it seems to trigger the second case. I'll add the second case to 'isTooLargeError'